bacon/model/StorageCouchBaseDB.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  StorageMongoDB.swift
+//  bacon
+//
+//  Underlying class of StorageManager to provide all saving and loading functionalities.
+//  Uses CouchBaseLiteSwfit to provide a full-featured embedded NoSQL database that runs locally.
+//
+//  Created by Travis Ching Jia Yea on 19/3/19.
+//  Copyright © 2019 nus.CS3217. All rights reserved.
+//
+
+import Foundation
+import CouchbaseLiteSwift
+
+private enum DatabaseCollections: String {
+    case transactions
+}
+
+class StorageCouchBaseDB {
+    private var transactionDatabase: Database
+
+    init() throws {
+        // Initialize database
+        transactionDatabase = try StorageCouchBaseDB.openOrCreateEmbeddedDatabase(name: .transactions)
+    }
+
+    private static func openOrCreateEmbeddedDatabase(name: DatabaseCollections) throws -> Database {
```

Comment:
I think you should include logging statements for this method, since it's relatively complex and have multiple points of failure. Will be good for debugging too.

Reply:
Will do.